Autopilot/PathManager/Inc/pathStateClasses.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include "pathManagerStateManager.hpp"
+#include "pathManager.hpp"
+#include "altimeter.hpp"
+#include "gps.hpp"
+#include "AutoSteer.hpp"
+#include "waypointManager.hpp"
+
+/***********************************************************************************************************************
+ * Code
+ **********************************************************************************************************************/
+
+class commsWithAttitude : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        commsWithAttitude() {}
+        commsWithAttitude(const commsWithAttitude& other);
+        commsWithAttitude& operator =(const commsWithAttitude& other);
+};
+
+class getFromTelemetry : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        getFromTelemetry() {}
+        getFromTelemetry(const getFromTelemetry& other);
+        getFromTelemetry& operator =(const getFromTelemetry& other);
+};
+
+class getSensorData : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static AltimeterData_t *GetAltimeterOutput(void) {return &_altimeterdata;}
+        static GpsData_t *GetGPSOutput(void) {return &_gpsdata;}
+    private:
+        getSensorData() {}
+        getSensorData(const getSensorData& other);
+        getSensorData& operator =(const getSensorData& other);
+        static AltimeterData_t _altimeterdata; 
+        static GpsData_t _gpsdata; 
+};
+
+class sensorFusion : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+    private:
+        sensorFusion() {}
+        sensorFusion(const sensorFusion& other);
+        sensorFusion& operator =(const sensorFusion& other);
+};
+
+class cruisingState : public pathManagerState
+{
+    public:
+        void enter(pathManager* pathMgr) {(void) pathMgr;}
+        void execute(pathManager* pathMgr);
+        void exit(pathManager* pathMgr) {(void) pathMgr;}
+        static pathManagerState& getInstance();
+        static _PathData *GetWaypointData(void) {return &_waypointdata;}
```

Comment:
I don't see any case where we would need this method... I would recommend removing it.